.github/actions/setup-project/action.yml

```diff
@@ -0,0 +1,32 @@
+name: setup project
+description: setup project by default installing dependencies and running setup script
+inputs:
+  setup:  # id of input
+    description: 'setup project'
+    required: true
+    default: 'true'
+runs:
+  using: "composite"
+  steps:
+    - name: Setup node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18
+
+    - name: Install pnpm
```

Comment:
you may use this action to install pnpm, others will be updated later
- uses: pnpm/action-setup@v2
        with:
          version: 8